source/PlayerInfo.cpp

```diff
@@ -1699,6 +1684,46 @@ bool PlayerInfo::TakeOff(UI *ui)
 
 
 
+void PlayerInfo::PoolCargo()
+{
+	// This can only be done while landed.
+	if(!planet)
+		return;
+	for(const shared_ptr<Ship> &ship : ships)
+		if(!ship->IsParked() && !ship->IsDisabled() && ship->GetPlanet() == planet)
+			ship->Cargo().TransferAll(cargo);
+}
```

Comment:
It can be used in land, but it was originally intended fo the case where the player aborts take off, but apparently I didn't add it there.

Reply:
Because there's no way to put it there - dialogs don't do callbacks on cancel, and that's what you need here.  Thus, #9122.  With that and a bit more code, this PR should work just fine.